### PR TITLE
Fix or remove TODOs

### PR DESCRIPTION
**Pull request by sisou**
There are TODOs left in the `QrEncoder.js`. @danimoh, is it possible to do what this TODO says
https://github.com/nimiq/keyguard-next/blob/master/src/lib/QrEncoder.js#L460:
```js
// TODO rm5 can be removed if we fix type to 5 (this method is called at 7 only)
```

### CODE REVIEW

File: tests/lib/KeyStore.spec.js [hunk 1]
```diff
@@ -157,7 +166,8 @@ describe('KeyStore', () => {
         const key1 = await KeyStore.instance._get(Dummy.keyInfos[0].id);
         expect(key1).toEqual(Dummy.storedKeyRecords()[0]);
 
-        // TODO: Expect Accounts DB to not be deleted
+        const accountsDb = await AccountStore.instance.connect();
+        expect(accountsDb).toBe(null);
```
Comment (Bettelstab) [inline tests/lib/KeyStore.spec.js]: Sure it should not be not.toBe? Can't take a deeper look right now.

File: src/lib/Utf8Tools.js [hunk 2]
```diff
@@ -10,13 +10,21 @@
...
 class Utf8Tools { // eslint-disable-line no-unused-vars
     /**
      * @param {string} str
      * @returns {Uint8Array}
      */
     static stringToUtf8ByteArray(str) {
-        // TODO: Use native implementations if/when available
+        if (typeof TextEncoder !== 'undefined') {
+            // @ts-ignore (Expected 0 arguments, but got 1.)
+            const encoder = new TextEncoder('utf-8'); // The encoding is a legacy argument.
```
[the remark this answers is not in the document]
Comment (sisou) [reply src/lib/Utf8Tools.js]: Who knows. I think it doesn't hurt to send it in.

File: src/lib/Utf8Tools.js [hunk 3]
```diff
@@ -49,7 +57,12 @@ class Utf8Tools { // eslint-disable-line no-unused-vars
      * @returns {string}
      */
     static utf8ByteArrayToString(bytes) {
-        // TODO: Use native implementations if/when available
+        if (typeof TextDecoder !== 'undefined') {
```
[the remark this answers is not in the document]
Comment (sisou) [reply src/lib/Utf8Tools.js]: I'd rather make it explicit. You never know when defaults might change xD

File: tests/lib/KeyStore.spec.js (hunk 1 above)
Comment (sisou) [reply tests/lib/KeyStore.spec.js]: The comment was actually wrong :D
That test really does the migration, and the test is mainly if it also removes the flag-cookies.

I added the `.not.toBe(null)` in the Iframe tests, where only the cookie should be set.

File: src/lib/Utf8Tools.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sisou) [reply src/lib/Utf8Tools.js]: However, the question is: was `utf-8` the default for the TextEncoder back then in those old browsers? If so, we can remove it, I guess.

[the remark this answers is not in the document]
Comment (sisou) [reply src/lib/Utf8Tools.js]: According to [this change of the spec](https://github.com/whatwg/encoding/commit/8360f775c8df145f649047c7d59c5ff733ade112#diff-1feda49b40370635faef8b655f144f64L1346), `utf-8` was already the default before it became the only encoding, so I will remove the parameter here.

**Pull request merged**